compiler/rustc_metadata/src/native_libs.rs

```diff
@@ -130,10 +131,11 @@ impl<'tcx> Collector<'tcx> {
                                 }
                                 NativeLibKind::RawDylib
                             }
+                            "link-arg" => NativeLibKind::LinkArg,
```

Comment:
`link-arg` is not supposed to be usable in `#[link]` attributes, only on the command line.

We need to keep the `src/test/run-make/pass-linker-flags-rs/main.rs` test in some form (in `src/test/ui` tests), but make sure that it produces an error.

Reply:
Could you restore the removed test `src/test/run-make/pass-linker-flags-rs/main.rs`?